test/e2e/util_test.go

```diff
@@ -472,17 +476,31 @@ func verifyExternalIngressController(t *testing.T, name types.NamespacedName, ho
 		}
 	}()
 
+	// If we have a DNS as an external IP address, make sure we can resolve it before moving on.
+	// This just limits the number of "could not resolve host" errors which can be confusing.
+	if net.ParseIP(address) == nil {
+		if err := wait.PollImmediate(10*time.Second, 5*time.Minute, func() (bool, error) {
+			_, err := net.LookupIP(address)
+			if err != nil {
+				t.Logf("waiting for loadbalancer domain %s to resolve...", address)
+				return false, nil
+			}
+			return true, nil
+		}); err != nil {
+			t.Fatalf("loadbalancer domain %s was unable to resolve:", address)
+		}
+	}
+
 	req, err := http.NewRequest(http.MethodGet, fmt.Sprintf("http://%s", address), nil)
 	if err != nil {
 		t.Fatalf("failed to build client request: %v", err)
 	}
-
 	// we use HOST header to map to the domain associated on the ingresscontroller.
 	// This ensures our http call is routed to the correct router.
 	req.Host = hostname
 
 	httpClient := http.Client{Timeout: 5 * time.Second}
-	err = waitForHTTPClientCondition(t, &httpClient, req, 10*time.Second, 5*time.Minute, func(r *http.Response) bool {
+	err = waitForHTTPClientCondition(t, &httpClient, req, 10*time.Second, 10*time.Minute, func(r *http.Response) bool {
 		if r.StatusCode == http.StatusOK {
```

Comment:
I think we should be draining the response buffer, ala: 

```go
io.Copy(ioutil.Discard, r.Body)
```

Reply:
Yup I agree, but `resp.Body.close()` does that for us if we don't do it ourselves: https://groups.google.com/g/golang-nuts/c/4Rr8BYVKrAI/m/HRY-N-8_orcJ

`resp.Body.close()` was added in `waitForHTTPClientCondition`.